Earthfile

```diff
@@ -39,241 +63,273 @@ setup:
-  RUN useradd -m ${USERNAME} && \
-    echo "${USERNAME}:${USERNAME}" | chpasswd && \
-    usermod --shell /bin/bash ${USERNAME} && \
-    usermod -aG sudo ${USERNAME} && \
-    mkdir -p /etc/sudoers.d && \
-    echo "${USERNAME} ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers.d/${USERNAME} && \
-    chmod 0440 /etc/sudoers.d/${USERNAME}
-  USER ${USERNAME}
-  WORKDIR ${SPACEROS_DIR}
-
-  # Update the OpenGL version
-  RUN sudo add-apt-repository ppa:kisak/kisak-mesa
-  RUN --mount=type=cache,mode=0777,target=/var/cache/apt,sharing=locked,id=cache_apt_cache \
-      --mount=type=cache,mode=0777,target=/var/lib/apt,sharing=locked,id=lib_apt_cache \
-      sudo apt update && sudo apt upgrade -y
+  RUN apt-get install -y python3-rosinstall-generator
 
-  # Create install location and copy in relevant scripts
-  COPY --chown ${USERNAME}:${USERNAME} --dir docker/scripts/ ${SPACEROS_DIR}
+  COPY scripts ./
+  COPY excluded-pkgs.txt spaceros-pkgs.txt spaceros.repos ./
 
-repos-file:
-  FROM +setup
-  COPY excluded-pkgs.txt ./
-  COPY spaceros-pkgs.txt ./
-  COPY spaceros.repos ./
   # This is a fresh image, so we do not need to exclude installed packages.
-  RUN --no-cache sh scripts/generate-repos.sh \
+  ENV AMENT_PREFIX_PATH=${SPACEROS_DIR}
+  RUN sh generate-repos.sh \
                  --outfile ros2.repos \
                  --packages spaceros-pkgs.txt \
                  --excluded-packages excluded-pkgs.txt \
-                 --exclude-installed false \
                  --rosdistro ${ROS_DISTRO}
-  RUN --no-cache python3 scripts/merge-repos.py ros2.repos spaceros.repos -o output.repos
+  RUN  python3 merge-repos.py ros2.repos spaceros.repos -o output.repos
   SAVE ARTIFACT output.repos AS LOCAL ros2.repos
 
-spaceros-artifacts:
-  # we must run it in a separate container, so that downstream tasks can be cached if vcs file does not change
-  FROM +setup
-  USER ${USERNAME}
-  WORKDIR ${SPACEROS_DIR}
-
-  COPY ros2.repos ./
-  COPY excluded-pkgs.txt ./
-
-  # this ensure the vcs import and export results are not cached
-  RUN --no-cache echo "Cloning spaceros repo artifacts"
-
-  # we run vcstool inside this task, because some packages in `ros2.repos` are not pinned and otherwise
-  # earthly won't pull latest changes
-  RUN mkdir src
-  RUN vcs import --shallow --retry 3 --input ros2.repos src
-  RUN vcs export --exact src > exact.repos
-
-  SAVE ARTIFACT ros2.repos
-  SAVE ARTIFACT exact.repos # `ros2.repos`, but with pinned versions (e.g. SHAs instead of branches)
-  SAVE ARTIFACT excluded-pkgs.txt
+###############################################################################
+### IKOS Installation
+# This stage is responsible for installing IKOS and its dependencies.
+# This will be used for static analysis of the ROS 2 workspace during build
+# test and development.
+###############################################################################
+ikos-install:
+  FROM +pre-installation
+
+  RUN apt-get install --yes \
```

Comment:
This will create additional layers in the image. I had to stuff them under a single command to remove the layers. 

As a note, I have trimmed the layers from 39 to 21 of the final image.

Reply:
Oh sorry, I just mean it's a little easier to read what's being installed when each package is on it's own line, ala:

```
  RUN apt-get install --yes \
      gcc \
      g++ \
      cmake \
      lib...
```

Reply:
Got you. I will update this.

Reply:
Looks good, thank you!